compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -2213,16 +2213,20 @@ class Typer extends Namer
             traverse(xtree :: rest, facts)
           case none =>
             import untpd.modsDeco
-            val ctx1 = if (ctx.explicitNulls) {
-              mdef match {
-                case mdef: untpd.ValDef if ctx.owner.is(Method) && !mdef.mods.isOneOf(Lazy | Implicit) =>
-                  ctx.fresh.addFlowFacts(facts)
-                case _ => ctx
-              }
-            } else {
-              ctx
+            val ctx2 = mdef match {
+              case mdef: untpd.ValDef
```

Comment:
Could you please add a comment explaining why we're doing this? You can partially re-use the old comment for ValDefInBlockCompleter